Python/sysmodule.c

```diff
@@ -3630,6 +3631,70 @@ make_impl_info(PyObject *version_info)
     return NULL;
 }
 
+
+static PyObject *
+make_abi_info(void)
+{
+    int res;
+    PyObject *abi_info, *value, *ns;
+    abi_info = PyDict_New();
+    if (abi_info == NULL) {
+        goto error;
+    }
+
+    value = PyLong_FromLong(sizeof(void *) * 8);
+    if (value == NULL) {
+        goto error;
+    }
+    res = PyDict_SetItemString(abi_info, "pointer_bits", value);
+    Py_DECREF(value);
+    if (res < 0) {
+        goto error;
+    }
+
+#ifdef Py_GIL_DISABLED
+    value = Py_True;
+#else
+    value = Py_False;
+#endif
+    res = PyDict_SetItemString(abi_info, "free_threaded", value);
+    if (res < 0) {
+        goto error;
+    }
+
+#ifdef Py_REF_DEBUG
+    value = Py_True;
+#else
+    value = Py_False;
+#endif
+    res = PyDict_SetItemString(abi_info, "debug", value);
+    if (res < 0) {
+        goto error;
+    }
+
+#if PY_BIG_ENDIAN
+    value = PyUnicode_FromString("big");
+#else
+    value = PyUnicode_FromString("little");
+#endif
+    if (value == NULL) {
+        goto error;
+    }
+    res = PyDict_SetItemString(abi_info, "byteorder", value);
+    if (res < 0) {
+        goto error;
+    }
+    Py_DECREF(value);
+
+    ns = _PyNamespace_New(abi_info);
+    Py_DECREF(abi_info);
+    return ns;
+
+error:
+    Py_CLEAR(abi_info);
```

Comment:
nitpick: there is not need to set *abi_info* to NULL.

```suggestion
    Py_DECREF(abi_info);
```